### PR TITLE
[NEW COMPONENT] NavLink

### PR DESCRIPTION
**Pull request by coltonehrman**
added NavLink component + converted <a> tags to NavLink

Added NavLink component to make it easier and cleaner for <a> tags in components
Was going to make the component throw an error when passed `internal` && `blank` but couldn't figure out how to test it properly

### CODE REVIEW

File: __tests__/components/NavLink.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+import { render } from '@testing-library/react'
+import NavLink from '../../components/NavLink'
+
+describe('NavLink Component', () => {
+  test('Should crash if given internal && blank props', () => {
+    // expect(() => render(<NavLink internal blank />)).toThrow()
```
Comment (songz) [inline __tests__/components/NavLink.js:7]: why do we need an empty test?

Comment (coltonehrman) [reply __tests__/components/NavLink.js:7]: In my comments I mentioned how I wasn't sure how to test this properly lol, I will either update it when I figure it out or remove it completely if we think it's not necessary

Comment (songz) [reply __tests__/components/NavLink.js:7]: storybook comment. Just render the different props is good enough.

Comment (coltonehrman) [reply __tests__/components/NavLink.js]: removed test

**Pull request closed without merging**